### PR TITLE
Fix/txt long len wrapping and disable macro

### PR DESCRIPTION
**Pull request by BrianPugh**
Fixes bug described in https://github.com/littlevgl/lvgl/issues/1256#issuecomment-552595634

I think between this and the last lv_txt PR, the logic is a lot cleaner and what I was originally going for. Fixing this last bug removed some funky logic.

### CODE REVIEW

**Comment by BrianPugh** [comment]
hold up, might have a slight oversight

**Comment by BrianPugh** [comment]
ok should be fine now, the line i added doesn't change any current functionality, it's just there for correctness.

**Comment by kisvegabor** [comment]
I've tried it and the original issue is solved indeed, however now the "pretty word wrap" seems not working:
```c
/* If a character is at least this long, will break wherever "prettiest" */
#ifndef LV_TXT_LINE_BREAK_LONG_LEN
#define LV_TXT_LINE_BREAK_LONG_LEN          7
#endif

/* Minimum number of characters of a word to put on a line before a break */
#ifndef LV_TXT_LINE_BREAK_LONG_PRE_MIN_LEN
#define LV_TXT_LINE_BREAK_LONG_PRE_MIN_LEN  3
#endif

/* Minimum number of characters of a word to put on a line after a break */
#ifndef LV_TXT_LINE_BREAK_LONG_POST_MIN_LEN
#define LV_TXT_LINE_BREAK_LONG_POST_MIN_LEN 5
#endif
```
long

**Comment by BrianPugh** [comment]
hm that only happens when it's a single long word taking up the entire line. Under all other cases it seems fine. I'll try and think of a solution.

**Comment by BrianPugh** [comment]
Fixed that bug, but in doing so I discovered one more bug: when clicking to the right of a wrapped long word, the cursor is one off. Its getting late, so i might have to fix that tomorrow.

**Comment by BrianPugh** [comment]
@kisvegabor It seems wrapping at break characters and clicking at the end of the line are at odds with eachother (I checked release v6.0 and the same bug is there). There would need to be a bigger change for cursors that I'm not comfortable making to resolve this problem. Most notably, in either system, you cannot put the cursor in front of the final break-char in a line (most commonly space). So to delete a break char at the end of a line, you have to delete it from the first cursor position on the next line. Essentially, a cursor should be able to represent the same string index but on different lines.

Here's a crude drawing illustrating that no index will put the cursor to the right of the 2nd "W" in the string represented by characters `{'W', 'W', 'W', '\0'}`

Capture

Somewhat related: In my previous PR, I added some code for last-line handling in `lv_label` `lv_label_get_letter_on` . I see a lot of similar code in `lv_label_is_char_under_pos`. Should the same code be put there? Or should these functions be refactored to remove duplicate code?

https://github.com/littlevgl/lvgl/pull/1259/files#diff-6c14ef100b87dc4aaacb5056073f96ccR664

In summary, in regards to cursor positioning, I believe this PR puts the txt wrapping on parity with the old wrapping system.

**Comment by kisvegabor** [comment]
I've tested the updates and the original issues seem solved.

> Most notably, in either system, you cannot put the cursor in front of the final break-char in a line (most commonly space).

If this bug existed in `v6.0` too then it's independent of the new line break code.  So we can merge your fix, and deal with this issue independently.

I've tested it in LibreOffice Writer and it was unintuitive there too:
break

---

Thank you for the fixes! :slightly_smiling_face:
